docs/DeveloperGuide.adoc

```diff
@@ -795,6 +883,20 @@ Private contact detail
 A contact detail that is not meant to be shared with others
 ....
 
+[[contact-list]]
+Contact List
+
+....
+A list that displays contacts that were added to it, a contact can exist on multiple lists
+....
+
+[[contact-history]]
+Contact History
+
+....
+Past interactions made with the contact person (e.g. email on 01-10-2017, CS2103 Updates)
+....
+
```

Comment:
It's good that the glossary is updated, but these terms aren't actually used in the dev guide.